src/main.ts

```diff
@@ -100,6 +110,177 @@ export class Minifier {
     return path.join(destination, subFilePath);
   }
 
+  isExternal(symbol: ts.Symbol): boolean {
+    // TODO: figure out how to deal with undefined symbols
+    // (ie: in case of string literal, or something like true.toString())
+    if (!symbol) return true;
+
+    return symbol.declarations.some((decl) => !!(decl.getSourceFile().fileName.match(/\.d\.ts/)));
+  }
+
+  isRenameable(symbol: ts.Symbol): boolean {
+    if (this.isExternal(symbol)) return false;
+    if (!this._typeCasting.has(symbol)) return true;
+
+    let boolArrTypeCasting = [];
+    let renameable = true;
+
+    // Three cases to consider:
+    // CANNOT RENAME: Use site passes an internally typed object, expected site wants an externally
+    // typed object
+    // CAN RENAME: Use site type symbols are internal, expected type symbols are internal
+    // ERROR: Expected symbol is external, use sites are both internal and external
+
+    // Create boolean array of whether or not actual sites (type to which a symbol is being cast)
+    // are internal
+    if (this._typeCasting.has(symbol)) {
+      for (let castType of this._typeCasting.get(symbol)) {
+        boolArrTypeCasting.push(!this.isExternal(castType));
+      }
+
+      // Check if there are both true and false values in boolArrTypeCasting, throw Error
+      if (boolArrTypeCasting.indexOf(true) >= 0 && boolArrTypeCasting.indexOf(false) >= 0) {
+        throw new Error(
+            'ts-minify does not support accepting both internal and external types at a use site');
+      }
+
+      // Check that all use sites are internal, if ALL internal, we can early return true, else
+      // false
+      for (let bool of boolArrTypeCasting) {
```

Comment:
since you already checked that the array has all trues or all falses, why not just `return boolArrTypeCasting[0]`